frontend/src/stores/user.ts

```diff
@@ -1,167 +1,165 @@
-                await router.push('/auth/login');
-            }
-        },
-        async routeLoggedIn() {
-            // @ts-ignore
-            const router = this.$router;
-            if (router.currentRoute.path === '/auth/login' || router.currentRoute.path === '/') {
-                await router.push('/main');
-            }
-        },
-        async passwordRecovery(payload: { userId: string }) {
-            const mainStore = useMainStore();
-            const loadingNotification = {content: 'Sending password recovery email', showProgress: true};
-            try {
-                mainStore.addNotification(loadingNotification);
-                await api.passwordRecovery(payload.userId);
-                mainStore.removeNotification(loadingNotification);
-                mainStore.addNotification({color: TYPE.SUCCESS, content: 'Password recovery link has been sent'});
-                await this.logout();
-            } catch (error) {
-                mainStore.removeNotification(loadingNotification);
-                let data = error.response.data;
-                let errMessage = "";
-                if (data.message === 'error.validation') {
-                    errMessage = data.fieldErrors.map(e => `${e.field}: ${e.message}`).join('\n');
-                } else {
-                    errMessage = data.detail;
-                }
-                mainStore.addNotification({color: TYPE.ERROR, content: errMessage});
-            }
-        },
-        async resetPassword(payload: { password: string, token: string }) {
-            const mainStore = useMainStore();
-            const loadingNotification = {content: 'Resetting password', showProgress: true};
-            mainStore.addNotification(loadingNotification);
-            await api.resetPassword(payload.password);
-            mainStore.removeNotification(loadingNotification);
-            mainStore.addNotification({color: TYPE.SUCCESS, content: 'Password successfully changed'});
-            await this.logout();
-        },
-        async signupUser(payload: { userData: ManagedUserVM }) {
-            const mainStore = useMainStore();
-            const loadingNotification = {content: 'saving', showProgress: true};
-            try {
-                mainStore.addNotification(loadingNotification);
-                await api.signupUser(payload.userData);
-                mainStore.removeNotification(loadingNotification);
-                mainStore.addNotification({content: 'Success! Please proceed to login', color: TYPE.SUCCESS});
-                await this.logout();
-            } catch (error) {
-                throw new Error(error.response.data.detail);
-            }
+          }
+        }
+      } else if (!this.isLoggedIn) {
+        const response = await api.getUserAndAppSettings();
+        this.isLoggedIn = true;
+        this.userProfile = response.user;
+        let appConfig = response.app;
+        mainStore.setAppSettings(appConfig);
```

Comment:
This lines are duplicated with lines 76-80. I think the order of if/else it can be improved to make it more concise

```typescript
if (!this.isLoggedIn) {
  if (mainStore.authAvailable) {
    this.token = this.token || getLocalToken();
    if (!token) {
      this.removeLogin();
      return;
    }
  }
  const response = await api.getUserAndAppSettings();
  this.isLoggedIn = true;
  this.userProfile = response.user;
  mainStore.setAppSettings(response.app);
}
```

(ideally we should break down to sub function to avoid having too much layers of if)

Reply:
I refactored this `checkLoggedIn` method.
Could you recheck it?

Reply:
I saw that you forgot about setting `this.token`

Reply:
Thanks!!